core/trino-main/src/main/java/io/trino/cost/ComparisonStatsCalculator.java

```diff
@@ -239,4 +250,93 @@ private static PlanNodeStatsEstimate estimateExpressionNotEqualToExpression(
+    {
+        if (leftExpressionStatistics.isUnknown() || rightExpressionStatistics.isUnknown()) {
+            return PlanNodeStatsEstimate.unknown();
+        }
+
+        StatisticRange leftRange = StatisticRange.from(leftExpressionStatistics);
+        StatisticRange rightRange = StatisticRange.from(rightExpressionStatistics);
+        if (leftRange.isEmpty() || rightRange.isEmpty()) {
+            return inputStatistics.mapOutputRowCount(rowCount -> 0.0);
+        }
+        // We don't know the correlation between NULLs, so we take the max nullsFraction from the expression statistics
+        // to make a conservative estimate for nullsFilterFactor
+        double nullsFilterFactor;
+        if (isNaN(leftExpressionStatistics.getNullsFraction()) && isNaN(rightExpressionStatistics.getNullsFraction())) {
+            nullsFilterFactor = 1.0;
+        }
+        else {
+            nullsFilterFactor = 1 - maxExcludeNaN(leftExpressionStatistics.getNullsFraction(), rightExpressionStatistics.getNullsFraction());
+        }
+
+        switch (operator) {
+            case LESS_THAN:
+            case LESS_THAN_OR_EQUAL:
+                // left is always lesser than right
+                if (leftRange.getHigh() < rightRange.getLow()) {
+                    PlanNodeStatsEstimate.Builder estimate = PlanNodeStatsEstimate.buildFrom(inputStatistics);
+                    leftExpressionSymbol.ifPresent(symbol -> estimate.addSymbolStatistics(
+                            symbol,
+                            leftExpressionStatistics.mapNullsFraction(nullsFraction -> 0.0)));
+                    rightExpressionSymbol.ifPresent(symbol -> estimate.addSymbolStatistics(
+                            symbol,
+                            rightExpressionStatistics.mapNullsFraction(nullsFraction -> 0.0)));
+                    return estimate.setOutputRowCount(inputStatistics.getOutputRowCount() * nullsFilterFactor)
+                            .build();
+                }
+                // left is always greater than right, no overlap
+                if (leftRange.getLow() > rightRange.getHigh()) {
+                    return inputStatistics.mapOutputRowCount(rowCount -> 0.0);
+                }
+                if (isNonEstimatablePredicateApproximationEnabled) {
+                    double overlappingRangeFraction = leftRange.overlapPercentWith(leftRange.intersect(rightRange));
+                    double alwaysLessRangeFraction = 0.0;
+                    if (leftRange.getLow() < rightRange.getLow()) {
+                        alwaysLessRangeFraction = leftRange.overlapPercentWith(new StatisticRange(leftRange.getLow(), rightRange.getLow(), NaN));
+                    }
+                    double filterFactor = alwaysLessRangeFraction + (UNKNOWN_FILTER_COEFFICIENT * overlappingRangeFraction);
+                    return inputStatistics.mapOutputRowCount(rowCount -> rowCount * filterFactor * nullsFilterFactor);
```

Comment:
You can derive symbol ranges:
I think it is
left: `[current_left_min, min(current_left_max, current_right_max)]`
right: `[max(current_right_min, current_left_min), current_right_max]`
You should adjust NDVs of symbols accordingly